### PR TITLE
Allow deleting a Snowflake connection

### PR DESCRIPTION
**Pull request by PopDaph**
## Description

This PR enable deletion of a Snowflake connection. 
This means both the new deletion modal + a hack to allow DELETING a datasource that is on system vault. 

Here's the flow. It follows the same design as the Oauth update flow. 
This is the Snowflake modal. We can see the button on top right is a new Delete button. 
<kbd>
Screenshot 2024-09-30 at 16 34 28
</kbd>

After clicking on it, we have a confirmation modal: 
<kbd>
Screenshot 2024-09-30 at 16 34 36
</kbd>

And then a confirmation: 
<kbd>
Screenshot 2024-09-30 at 16 35 12
</kbd>

## Risk

Can be rolled back. 

## Deploy Plan

Deploy front 
Deploy connector
(no specific order required).

### CODE REVIEW

File: connectors/src/connectors/snowflake/index.ts [hunk 1]
```diff
@@ -80,11 +80,11 @@ export class SnowflakeConnectorManager extends BaseConnectorManager<null> {
   }
 
   async clean(): Promise<Result<undefined, Error>> {
-    throw new Error("Method clean not implemented.");
+    return new Ok(undefined);
```
Comment (PopDaph) [inline connectors/src/connectors/snowflake/index.ts:83]: Will be implemented when we have the workflow (in progress will be done today), in the meantime we want to be able to delete it and not throw this error.

File: front/pages/api/w/[wId]/vaults/[vId]/data_sources/[dsId]/index.ts [hunk 2]
```diff
@@ -126,6 +125,23 @@ async function handler(
       });
     }
     case "DELETE": {
+      const isAuthorized =
+        vault.canWrite(auth) ||
```
Comment (spolu) [inline front/pages/api/w/[wId]/vaults/[vId]/data_sources/[dsId]/index.ts:129]: This one might be false?

**Review by fontanierh — APPROVED** [review]
LGTM

File: front/pages/api/w/[wId]/vaults/[vId]/data_sources/[dsId]/index.ts (hunk 2 above)
Comment (PopDaph) [reply front/pages/api/w/[wId]/vaults/[vId]/data_sources/[dsId]/index.ts:129]: I don't understand the question, can you clarify? 🙏🏻

Comment (spolu) [reply front/pages/api/w/[wId]/vaults/[vId]/data_sources/[dsId]/index.ts:129]: Sorry. I meant that I'm not sure admins can write here? Checking that it is an admin is enough as this is administration of the vault.

But I guess we decided that can write is true for admins 👍

Comment (PopDaph) [reply front/pages/api/w/[wId]/vaults/[vId]/data_sources/[dsId]/index.ts:129]: Admin canWrite is false for admins on the system vault, hence this workaround! (Or did I missed something?)

Comment (spolu) [reply front/pages/api/w/[wId]/vaults/[vId]/data_sources/[dsId]/index.ts:129]: Yes right :+1:

I think we should introduce a `canAdmin` to use explicitely for all admin operations. Current code LGTM

**Pull request merged**